src/common/function/src/scalars/json/json_get.rs

```diff
@@ -172,41 +175,211 @@ impl Function for JsonGetString {
-            let json = jsons.is_valid(i).then(|| jsons.value(i));
-            let path = paths.is_valid(i).then(|| paths.value(i));
-            let result = match (json, path) {
-                (Some(json), Some(path)) => {
-                    get_json_by_path(json, path).and_then(|json| jsonb::to_str(&json).ok())
-                }
-                _ => None,
+        Ok(ColumnarValue::Array(result))
+    }
+}
+
+fn jsonb_get_string(jsons: &BinaryViewArray, paths: &StringViewArray) -> Result<ArrayRef> {
+    let size = jsons.len();
+    let mut builder = StringViewBuilder::with_capacity(size);
+
+    for i in 0..size {
+        let json = jsons.is_valid(i).then(|| jsons.value(i));
+        let path = paths.is_valid(i).then(|| paths.value(i));
+        let result = match (json, path) {
+            (Some(json), Some(path)) => {
+                get_json_by_path(json, path).and_then(|json| jsonb::to_str(&json).ok())
+            }
+            _ => None,
+        };
+        builder.append_option(result);
+    }
+
+    Ok(Arc::new(builder.finish()))
+}
+
+fn json_struct_get_string(jsons: &StructArray, paths: &StringViewArray) -> Result<ArrayRef> {
+    let size = jsons.len();
+    let mut builder = StringViewBuilder::with_capacity(size);
+
+    for i in 0..size {
+        if jsons.is_null(i) || paths.is_null(i) {
+            builder.append_null();
+            continue;
+        }
+        let path = paths.value(i);
+
+        // naively assume the JSON path is our kind of indexing to the field, by removing its "root"
+        let field_path = path.replace("$.", "");
+        let column = jsons.column_by_name(&field_path);
+
+        if let Some(column) = column {
+            if let Some(v) = string_array_value_at_index(column, i) {
+                builder.append_value(v);
+            } else {
+                builder.append_value(arrow_cast::display::array_value_to_string(column, i)?);
+            }
+        } else {
+            let Some(raw) = jsons
+                .column_by_name(JsonStructureSettings::RAW_FIELD)
+                .and_then(|x| string_array_value_at_index(x, i))
+            else {
+                builder.append_null();
+                continue;
             };
-            builder.append_option(result);
+
+            let path: JsonPath<Value> = JsonPath::try_from(path).map_err(|e| {
+                DataFusionError::Execution(format!("{path} is not a valid JSON path: {e}"))
+            })?;
+            // the wanted field is not retrievable from the JSON struct columns directly, we have
+            // to combine everything (columns and the "_raw") into a complete JSON value to find it
+            let value = json_struct_to_value(raw, jsons, i)?;
+
+            match path.find(&value) {
+                Value::Null => builder.append_null(),
+                Value::Array(values) => match values.as_slice() {
+                    [] => builder.append_null(),
+                    [x] => {
```

Comment:
Any reference for this behaviour, treating single element array as its inner value

Reply:
it's documented in the `JsonPath::find`: the returned value must be either "null" or a list of `Value`